### PR TITLE
feat: add AMKA number validation function

### LINKED ISSUE #40: ValidationUtils: add validateΑΜΚΑ function
Add `validateΑΜΚΑ` function to `validationUtils.ts` module.

### PR DESCRIPTION
**Pull request by karate**
Closes #40 

Validates the Greek social security number ("Αριθμός Μητρώου Κοινωνικής Ασφάλισης" aka ΑΜΚΑ)

ΑΜΚΑ is a 11 digit number.
The first 6 digits is the date-of-birth in DDMMYY format.
Another 4-digit auto-increment number follows.
A checksum is added at the end, calculated using the [Luhn algorithm](https://en.wikipedia.org/wiki/Luhn_algorithm).

### CODE REVIEW

File: src/validationUtils.ts
Comment (raptisj) [inline src/validationUtils.ts:23]: @karate I believe you mean computed here instead of "cimputed"

Comment (karate) [reply src/validationUtils.ts]: Indeed. Thanks.

File: src/validationUtils.ts
Comment (tsevdos) [inline src/validationUtils.ts:53]: For consistency,  let's do `===`. (I 'll add it to the linter rules as well) 

```suggestion
  return checksum === parseInt(strAmka.charAt(10));
```

**Review by tsevdos — CHANGES_REQUESTED** [review]

File: src/validationUtils.ts
Comment (tsevdos) [reply src/validationUtils.ts]: Let's get a final approval (maybe from @vorillaz or @raptisj raptisj ) regarding the `Luhn algorithm` implementation and I'll be more than happy to merge it. 

Finally, please don't forget to export the `validateAMKA` to [index.ts](https://github.com/tsevdos/elUtils/blob/main/src/index.ts)

**Review by tsevdos — APPROVED** [review]

**Comment by tsevdos** [comment]
Let's get a final approval from another person (feel free to suggest one) regarding the Luhn algorithm implementation and I'll be more than happy to merge it.

File: src/validationUtils.ts
Comment (karate) [inline src/validationUtils.ts:51]: I think this is useless, because `checksum` will always be less than `9`

File: src/validationUtils.ts
Comment (karate) [inline src/validationUtils.ts:44]: This could be re-written as
```js
// sum digits of a 2-digit number
d = d - 9
```
It's simpler and faster, but mayby not so easy to read. What do you think?

Comment (tsevdos) [reply src/validationUtils.ts:44]: Maybe check this implementation https://stackoverflow.com/questions/12310837/implementation-of-luhn-algorithm and get another approval before merging.

Comment (raptisj) [reply src/validationUtils.ts]: @tsevdos @karate I find the above to be a good solution. The only thing I would add(nitpick) is more comments so we don't have to reread the Wikipedia resource constantly. But of course, you can skip this 🙂

**Review by raptisj — APPROVED** [review]

File: src/validationUtils.ts
Comment (karate) [reply src/validationUtils.ts]: Are you thinking of an overview of the algorithm before the reduce function, or a step-by-step explanation inside the loop?

Comment (raptisj) [reply src/validationUtils.ts]: @karate Your last commit is exactly what I had in mind

Comment (tsevdos) [reply src/validationUtils.ts]: Thank you both so much! I'm merging it 😁

**Pull request merged**
